### ISSUE TITLE
Wildcard imports and IntelliJ do not play well together.

### ISSUE THREAD

**Issue by bolinfest** [Kotlin/kotlinx.serialization#428]
I have the following code:

```
// Example.kt

package com.example

import kotlinx.serialization.*
import kotlinx.serialization.json.Json

@Serializable
data class Foo(val foo: String)

fun yay() {
  val result = Json.parse(Foo.serializer().list, """["foo", "bar"]""")
  println(result)
}
```

I can compile this with the following command line invocation:

```
kotlinc -classpath kotlinx-serialization-runtime-0.10.0.jar -Xplugin=kotlinx-serialization-compiler-plugin.jar Example.kt
```

However, if this line:

```
import kotlinx.serialization.*
```

is changed to:

```
import kotlinx.serialization.Serializable
```

and I try to compile it using the same command line invocation, I get the following error:

```
Example.kt:10:44: error: unresolved reference: list
  val result = Json.parse(Foo.serializer().list, """["foo", "bar"]""")
```

In IntelliJ, I have two problems with this.

First, the "organize imports" feature rewrites the wildcard import to `import kotlinx.serialization.Serializable`, which breaks things.

Second, even if I disable "organize imports," IntelliJ still fails to compile my file, citing an "unresolved reference" on `serializer()`. I have gone to **Build, Execution, Deployment > Compiler > Kotlin Compiler** and added `-Xplugin=/tmp/kotlinx-serialization-compiler-plugin.jar` to **Additional command line parameters**, but it still does not fix the build issue in IntelliJ.

Note that my IntelliJ project does not build with Maven or Gradle (this is for the Buck codebase itself), so the standard setup instructions on https://github.com/Kotlin/kotlinx.serialization do not work for me. I would have expected my change to the **Kotlin Compiler** settings to be sufficient.

This is with IntelliJ 2018.3, community edition. Under **Languages & Frameworks > Kotlin**, it says I am using 1.3.21-release-IJ2018.3-1.

**Comment by sandwwraith** [comment]
`.list` is an [extension function](https://github.com/kotlin/kotlinx.serialization/blob/c5bca9394b570289c525f772f81beaf2843f8554/runtime/common/src/main/kotlin/kotlinx/serialization/Shorthands.kt#L21) on KSerializer, so it should be imported separately if you do not use wildcards. If intellij IDEA does not recognize this function being used and does not import it on 'organize imports', it is either IDEA or Kotlin plugin bug. 

I'm not sure if compiler plugins are even supported in the IntelliJ internal build system without model importing from Maven/Gradle. Probably specifying plugin in **Project Settings** in Modules or Facets section could help

**Comment by bolinfest** [comment]
Ah, thanks! Indeed changing it to explicit exports as follows works when I compile from the command line:

```
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.list
```

I tried your suggestions for the IntelliJ issue. I created a Kotlin facet for the module within my project that contains the Kotlin code. It seems to present the same options as that of the top-level compiler, so I added `-Xplugin=kotlinx-serialization-compiler-plugin.jar` again, but it did not eliminate the build error when editing/saving in IntelliJ.

I also added `import kotlinx.serialization.list`, but when I saved the file, organize imports removed it again.

The one thing I could do was edit the file outside of IntelliJ in vim, add `import kotlinx.serialization.list`, save it, go back to IntelliJ, and then do `cmd-shift-f9` to recompile only `Example.kt`. No errors appeared in the "rebuild" pane, but `serializer` still appears in red in the editor and mousing over it says "Unresolved reference: serializer."

Indeed, Kotlin plugins appear to require special treatment for IntelliJ to recognize them correctly. I assume my best-case scenario is that there is some undocumented XML element I can add to my `.iml` file under `<facet type="kotlin-language" name="Kotlin">` that will let me specify the plugin?
